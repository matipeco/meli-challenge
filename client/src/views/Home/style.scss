.landing{
  &__container{
    & > div{
      display: flex;
      align-items: center;
      justify-content: center;
    }
    padding: 0;
  }
  &__background{
    // width: 100%;
    height: 90vh;
  }
}