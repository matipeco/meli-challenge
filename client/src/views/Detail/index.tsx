import { useParams } from "react-router-dom";
import { Container } from "../../components/Container"
import { useEffect, useState } from "react";
import axios from "axios";
import { DetailProduct, DetailResponse } from "../../types";
import { Breadcrumb } from "../../components/Breadcrumb";
import './style.scss';

export const Detail = () => {
  const { id } = useParams();

  const [product, setProduct] = useState<DetailProduct>();
  const [categories, setCategories] = useState<string[]>([]);
  

  const getDetail = async() => {
    const { data } = await axios.get<DetailResponse>(`http://localhost:3001/api/items/${id}`);
    setProduct(data.item)
    setCategories(data.categories)
  }

  useEffect(()=>{
    getDetail();
  },[])
  
  if(!product) return null;
  
  const priceFormat = product.price.amount.toLocaleString('es-AR', {
    style: 'currency',
    currency: product.price.currency,
    minimumFractionDigits: 0,
  })

  const getDecimals = (decimals:number) =>{
    if(decimals === 0){
      return "00"
    }
    else if(decimals < 10 ){
      return `0${decimals}`
    }
    return decimals;
  }

  const descriptionArray = product.description.split("\n").filter((str)=> str);

  return(
    <main>
      <Container>
        <Breadcrumb categories={categories}/>
        <article className="detail__container">
          <img className="detail__image" src={product.picture} alt={product.title} />
          
          <div className="detail__contain">
            <p className="detail__condition">{product.condition} - {product.sold_quantity} {product.sold_quantity === 1 ? "vendido" : "vendidos"} </p>
            <h2 className="detail__title">
              {product.title}
            </h2>
            <p className="detail__price">{priceFormat}
              <sup>{getDecimals(product.price.decimals)}</sup> 
            </p>
            <button className="detail__button">Comprar</button>
          </div>
          <h3 className="detail__description-title">Descripci√≥n del producto</h3>
          {descriptionArray.map((paragraph)=>{
            return(
              <p className="detail__description-text" key={paragraph}>{paragraph}</p>
            )
          })}
        </article>
      </Container>
    </main>
  )
}