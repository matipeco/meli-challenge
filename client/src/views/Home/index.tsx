import { Container } from "../../components/Container"
import './style.scss'
import landing from '../../assets/meli-background.png'

export const Home =()=>{
  return(
    <main className="landing__container">
      <Container>
        <img className="landing__background" src={landing} alt="mercadolibre landing" width="500" />
      </Container>
    </main>
  )
}