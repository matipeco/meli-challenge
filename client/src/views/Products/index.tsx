import { FunctionComponent, useEffect, useState } from "react"
import { Container } from "../../components/Container"
import axios from 'axios';
import { SearchResponse } from "../../types";
import { useLocation } from "react-router-dom";
import { Card } from "../../components/Card";
import './style.scss'

export const Products:FunctionComponent = () => { 
  const { search } = useLocation();
  const params = new URLSearchParams(search);
  const q = params.get('q');

  const [info, setInfo] = useState<SearchResponse>();

  const getProducts = async() =>{
    try {
      const { data } = await axios.get<SearchResponse>(`http://localhost:3001/api/items?q=${q}`);
      setInfo(data);
      console.log(data)

    } catch (error) {
      console.log(error);
    }
  }

  useEffect( ()=>{
    getProducts()
  },[q])

  {console.log(info?.categories)}
  return(
   <main className="container-cards">
    <Container>
        {info?.categories.map((cat)=>{
          return(
            <span className="categories">{cat} - </span>
          )
        })}
      {info?.items.map((prod)=>{
        return(
          <Card prod={prod}/>
        )
      })}
    </Container>
   </main> 
  )
}