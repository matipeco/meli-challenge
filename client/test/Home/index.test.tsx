import { render } from '@testing-library/react'
import { Home } from '../../src/views/Home'

describe("Componente Home", ()=>{
  test("Se renderiza correctamente", ()=>{
    const {container} = render(
      <Home/>
    )
    expect(container).toMatchSnapshot();
  })
})