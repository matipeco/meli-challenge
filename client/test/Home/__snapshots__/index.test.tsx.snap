// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Componente Home > Se renderiza correctamente 1`] = `
<div>
  <main
    class="landing__container"
  >
    <div
      class="container"
    >
      <img
        alt="mercadolibre landing"
        class="landing__background"
        src="/src/assets/meli-background.png"
        width="500"
      />
    </div>
  </main>
</div>
`;
